### PR TITLE
Feature/controller

### PR DESCRIPTION
**Pull request by NicolaiFransen**
Creates the PI objects and calcualtes the current reference values

### CODE REVIEW

**Review by AitorTeran — APPROVED** [review]
Looks good¡¿

File: Software/Development/SemesterProject/App/controller.c [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+static struct
+{
+    PIobject IqController;
+    PIobject IdController;
+    PIobject SpeedController;
+} PIControllerList;
+
+void initPIControllers(void)
+{
+    PIObject_Constructor(&PIControllerList.IdController, KP_ID, KI_ID, CURRENT_LIMIT, 0);
```
Comment (murgui) [inline Software/Development/SemesterProject/App/controller.c:24]: image

I like constructor's variable names 👍

**Review by murgui — CHANGES_REQUESTED** [review]
Nice job.

I think there should be an external interface for resetting integrators. This interface would be used when an error is detected, then after the user ACK, the system will start from scratch.

**Comment by NicolaiFransen** [comment]
I have created the functions to reset the integrators. 
For now im calling it while in the system is in standby state. Then we do the reset both when an error happens and when you change the reference.

**Review by murgui — APPROVED** [review]
¡¡¡¡SHIP IT!!!!

**Pull request merged**